### PR TITLE
Support Expression build in AstBuilder

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wangruin** [comment]
run starrocks_be_unittest

**Comment by wangruin** [comment]
run starrocks_be_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -338,4 +1100,118 @@ private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
...
+                return Type.CHAR;
+            } else if (signature.equalsIgnoreCase("STRING")) {
+                ScalarType stringType = ScalarType.createVarcharType(ScalarType.DEFAULT_STRING_LENGTH);
+                stringType.setAssignedStrLenInColDefinition();
+                return stringType;
+            } else if (signature.equalsIgnoreCase("BITMAP")) {
+                return Type.BITMAP;
+            } else if (signature.equalsIgnoreCase("HLL")) {
+                return Type.HLL;
+            } else if (signature.equalsIgnoreCase("PERCENTILE")) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1172]: JSON type is ignored

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1172]: How about move this method to Type.java, to make the code of type system more compact?

**Review by murphyatwork — COMMENTED** [review]
Any test case for this PR?

**Comment by HangyuanLiu** [comment]
> Any test case for this PR?

There is currently no independent UT, and all code has no logical entry. I will change the Parser entry in all UTs to the new Parser in the next PR that opens the new Parser. At that point, the new Parser can use all existing tests

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1172]: > JSON type is ignored

When I was developing, JSON TYPE was also under development. So it is not included. I don't want to introduce JSON TYPE support in this PR anymore. I can support JSON TYPE in the next separate PR.

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1172]: > How about move this method to Type.java, to make the code of type system more compact?

I totally agree with this idea. But there's too much confusion in Type at the moment, and I don't think now is the right time to make it more complicated. For example, the `public static ScalarType createType(String type) ` function in ScalarType generates Type from a string, but its definition is not clear and cannot meet the actual needs of Parser. So I don't want to abstract a type generator function that is only used in Parser into Type in this PR.

**Review by murphyatwork — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**